Clock: 12 hour "midnight format"

956MB opened the pull request:
# What's new

I agree with @GonzalesGabe in #317 that this could probably be an option for users to select their preference on how the 12hr clock is displayed.

Adds the options `12:XX` and `00:XX` in `Settings > System > Midnight Format` to display the preferred format past midnight on all clocks (Desktop, Main Menu MNTM style, Lock screen and Nightstand clock app). "12:30 AM" -> "12:30" OR "00:30".

*"Midnight Format" could also be changed to something like "12AM Format" to eliminate the scrolling.*

#### 12:XX

![image (11)](https://github.com/user-attachments/assets/4ef5caa5-8584-46c3-a1f9-9fd89418d7e6)

#### 00:XX

![image (14)](https://github.com/user-attachments/assets/9e90e577-bf63-4d15-9910-8b65e80dd6a9)

-----
# For the reviewer

- [x] I've uploaded the firmware with this patch to a device and verified its functionality
- [x] I've confirmed the bug to be fixed / feature to be stable

956MB committed:
Clock: 12 hour midnight format

- References #317 and adds the options `12:XX` and `00:XX` in `Settings > System > Midnight Format` to display the preferred clock format past midnight on all clocks (Desktop, Main Menu MNTM style, Lock screen and Nightstand clock app). "12:30 AM" -> "00:30" OR "12:30".

WillyJL reviewed (changes requested):
RTC is "sacred" xD

its limited and will persist between firmwares, bad idea to add things here if not also added to OFW

i'd say propose to OFW and see what they think, if they shut it down then we can just use momentum settings

956MB committed:
Fix: Move midnight format setting out of furi_rtc

- Also felt like the midnight format setting was too out of place in `MNTM > Interface` with the 5 other submenu entries, so I put it in Misc for now.

956MB committed:
Merge branch 'dev' into feat/midnight-clock-format

956MB committed:
clock app external

956MB committed:
Merge branch 'dev' into feat/midnight-clock-format

# Conflicts:
#	applications/external

956MB committed:
Moved midnight format setting into new `Interface > General`

WillyJL commented on applications/main/momentum_app/scenes/momentum_app_scene_interface_general.c:
```diff
@@ -0,0 +1,61 @@
+static void momentum_app_scene_interface_general_midnight_format_changed(VariableItem* item) {
+    MomentumApp* app = variable_item_get_context(item);
+    bool value = variable_item_get_current_value_index(item);
+    variable_item_set_current_value_text(item, value ? "00:XX" : "XX:00");
```
im assuming this is supposed to be
```suggestion
    variable_item_set_current_value_text(item, value ? "00:XX" : "12:00");
```

WillyJL committed:
Merge remote-tracking branch 'mntm/dev' into mntm-pr-341

WillyJL committed:
Update applications/main/momentum_app/scenes/momentum_app_scene_interface_general.c

WillyJL committed:
Merge remote-tracking branch 'mntm/dev' into mntm-pr-341

WillyJL committed:
Merge remote-tracking branch 'mntm/dev' into mntm-pr-341

WillyJL committed:
Update changelog
